programs/fileio.c

```diff
@@ -422,6 +424,10 @@ void FIO_setTargetCBlockSize(FIO_prefs_t* const prefs, size_t targetCBlockSize)
     prefs->targetCBlockSize = targetCBlockSize;
 }
 
+void FIO_setSrcSizeHint(FIO_prefs_t* const prefs, size_t srcSizeHint) {
```

Comment:
We should take `MIN((size_t)INT_MAX, srcSizeHint)` here. Otherwise if someone passes a large source size hint, we will fail the compression.

Reply:
Makes sense. Should I change that definition to an `int` as well? I didn't realize the bounds were specified as ints when writing this originally.

Reply:
I'm going to change all instances of `srcSizeHint` to be ints actually. There's really no reason for it to be a size_t